### PR TITLE
Refactored Adyen Giving

### PR DESCRIPTION
**Pull request by zenit2001**
This PR contains the refactoring of Adyen Giving code.

### CODE REVIEW

File: src/cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js [hunk 1]
```diff
@@ -56,16 +56,16 @@ server.post(
  * Complete a donation through adyenGiving
  */
 server.post('Donate', server.middleware.https, (req /* , res, next */) => {
-  const { pspReference } = req.form;
   const { orderNo } = req.form;
+  const { orderToken } = req.form;
```
Comment (amihajlovski) [inline src/cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js:60]: you can put orderToken inside the brackets in line 59

File: src/cartridges/int_adyen_overlay/cartridge/scripts/adyenGiving.js [hunk 2]
```diff
@@ -54,13 +63,25 @@ function donate(donationReference, donationAmount, originalReference) {
       shopperInteraction: constants.SHOPPER_INTERACTIONS.CONT_AUTH,
     };
 
+    if (availableDonationAmounts.indexOf(parseInt(donationAmount.value)) === -1){
+      throw new Error(`Donation amount is invalid`);
+    };
```
Comment (amihajlovski) [inline src/cartridges/int_adyen_overlay/cartridge/scripts/adyenGiving.js:68]: colon is not needed here

File: src/cartridges/int_adyen_overlay/cartridge/scripts/adyenGiving.js [hunk 3]
```diff
@@ -54,13 +63,25 @@ function donate(donationReference, donationAmount, originalReference) {
...
+
+    if (paymentCurrency !== donationAmount.currency){
```
Comment (amihajlovski) [inline src/cartridges/int_adyen_overlay/cartridge/scripts/adyenGiving.js:70]: looks like an issue with linting again

File: src/cartridges/int_adyen_overlay/cartridge/scripts/adyenGiving.js [hunk 4]
```diff
@@ -54,13 +63,25 @@ function donate(donationReference, donationAmount, originalReference) {
...
+      throw new Error(`Donation currency is invalid`);
+    };
```
Comment (amihajlovski) [inline src/cartridges/int_adyen_overlay/cartridge/scripts/adyenGiving.js:72]: colon not needed

File: src/cartridges/int_adyen_overlay/cartridge/scripts/adyenGiving.js (hunk 3 above)
Comment (zenit2001) [reply src/cartridges/int_adyen_overlay/cartridge/scripts/adyenGiving.js:70]: It seems like linter is not checking all the files

**Review by zaiddreakh — APPROVED** [review]

**Review by amihajlovski — APPROVED** [review]

**Pull request merged**